tests/test_some_of.py

```diff
@@ -0,0 +1,191 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import unittest
+
+from parameterized import parameterized
+
+from monai.data import MetaTensor
+from monai.transforms import TraceableTransform, Transform
+from monai.transforms.compose import Compose, SomeOf
+from monai.utils import set_determinism
+from monai.utils.enums import TraceKeys
+from tests.test_one_of import NonInv
+from tests.test_random_order import InvC, InvD
+
+
+class A(Transform):
+    def __call__(self, x):
+        return 2 * x
+
+
+class B(Transform):
+    def __call__(self, x):
+        return 3 * x
+
+
+class C(Transform):
+    def __call__(self, x):
+        return 5 * x
+
+
+class D(Transform):
+    def __call__(self, x):
+        return 7 * x
+
+
+KEYS = ["x", "y"]
+TEST_COMPOUND = [
+    (SomeOf((A(), B(), C()), fixed=True, max_num_transforms=3), 2 * 3 * 5),
+    (Compose((SomeOf((A(), B(), C()), fixed=True, max_num_transforms=3), D())), 2 * 3 * 5 * 7),
+    (SomeOf((A(), B(), C(), Compose(D())), fixed=True, max_num_transforms=4), 2 * 3 * 5 * 7),
+]
+
+# Modified from RandomOrder
+TEST_INVERSES = [
+    (SomeOf((InvC(KEYS), InvD(KEYS))), True, True),
+    (Compose((SomeOf((InvC(KEYS), InvD(KEYS))), SomeOf((InvD(KEYS), InvC(KEYS))))), True, False),
+    (SomeOf((SomeOf((InvC(KEYS), InvD(KEYS))), SomeOf((InvD(KEYS), InvC(KEYS))))), True, False),
+    (SomeOf((Compose((InvC(KEYS), InvD(KEYS))), Compose((InvD(KEYS), InvC(KEYS))))), True, False),
+    (SomeOf((NonInv(KEYS), NonInv(KEYS))), False, False),
+]
+
+
+class TestSomeOf(unittest.TestCase):
```

Comment:
The code coverage checker is flagging a few corner cases still. Once those are resolved, I'll do a proper pass over the tests